packages/studio-plugin/src/ParsingOrchestrator.ts

```diff
@@ -155,34 +178,52 @@ export default class ParsingOrchestrator {
 
   private calculateStudioData(): StudioData {
     const siteSettings = this.getSiteSettings();
-    const pageRecords = Object.keys(this.pageNameToPageFile).reduce(
+    const layoutNameToLayoutMetadata = this.constructRecordsFromSourceFile(
+      this.layoutNameToLayoutFile,
+      (layoutFile) => layoutFile.getLayoutMetadata()
+    ).nameToSuccessData;
+    const pageRecords = this.constructRecordsFromSourceFile(
+      this.pageNameToPageFile,
+      (pageFile) => pageFile.getPageState()
+    );
+
+    return {
+      pageNameToPageState: pageRecords.nameToSuccessData,
+      pageNameToErrorPageState: pageRecords.nameToError,
+      layoutNameToLayoutMetadata,
+      UUIDToFileMetadata: this.getUUIDToFileMetadata(),
+      siteSettings,
+      studioConfig: this.studioConfig,
+      isWithinCBD: !!process.env.YEXT_CBD_BRANCH,
+    };
+  }
+
+  private constructRecordsFromSourceFile<SourceFile, SuccessData>(
```

Comment:
@oshi97, this method feels a bit confusing to me, especially in it's return type. I wonder if there's a better way to consolidate this logic between Layouts & Pages.

Reply:
renamed to separateErrorAndSuccessResults! Not sure about these names but I think they're ok for now